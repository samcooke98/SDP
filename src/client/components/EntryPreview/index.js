import React, { Component } from 'react';
import style from "./web.css"

import { Link } from "react-router-dom"


export default class EntryPreview extends Component {
    render() {
        return (
            <div className={style.card}>
                <h3> {this.props.title} - {this.props.date} </h3>
                <span className={style.content} dangerouslySetInnerHTML={({ __html: this.props.preview.replace("<p>&nbsp;</p>", '') })} />

                <Link to={this.props.to} onClick={this.props.onClick}> Open Entry </Link>
            </div>
        );
    }
}

function createMarkup() {
    return { __html: 'First &middot; Second' };
}
//TODO: Prevent XSS 

/*
<Link to={`/journal/${id}/${entry}/${val}`}>
                                            {val} <br />
                                        </Link>*/
