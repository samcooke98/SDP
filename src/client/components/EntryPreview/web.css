

.card { 
    width: 100%;
    background-color: #F8F8F8;
    box-sizing: border-box;
    padding: 4px;
    margin-bottom: 12px;

    height: 150px; 
    overflow: hidden;
    display: flex;
    flex-direction: column;
}

.content { 
    flex-shrink: 1;
    overflow: hidden;
    text-overflow: ellipsis;
}

.content > p:first-child { 
    margin-top: 0;
    height: 100%; 
    text-overflow: ellipsis;
    
}

.card > a { 
    margin-top: 3px;
    margin-bottom: 3px;
    margin-left: auto; 
}