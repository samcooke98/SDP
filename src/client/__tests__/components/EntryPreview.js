import React from "react";
import renderer from "react-test-renderer";
// import ReactDOM from 'react-dom';
import { MemoryRouter } from "react-router-dom";
import EntryPreview from "../../components/EntryPreview/index";


describe('React Component: EntryPreview', () => {
    it("matches a snapshot", () => {
        const tree = renderer.create(
            <MemoryRouter>
                <EntryPreview />
            </MemoryRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
})