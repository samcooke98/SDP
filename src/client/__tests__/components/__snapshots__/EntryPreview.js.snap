// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React Component: EntryPreview matches a snapshot 1`] = `
<div
  className={undefined}
>
  <h3>
     
     - 
     
  </h3>
  <span
    className={undefined}
    dangerouslySetInnerHTML={
      Object {
        "__html": "Hello World!",
      }
    }
  />
  <a
    href="/"
    onClick={[Function]}
  >
     Open Entry 
  </a>
</div>
`;
